tests/test_bootstrap.py

```diff
@@ -92,9 +92,16 @@ def test_create_replica_old_format(self, mock_cancellable_subprocess_call):
         mock_cancellable_subprocess_call.return_value = 1
         self.assertEqual(self.b.create_replica(self.leader), 1)
 
-    def test_basebackup(self):
-        self.p.cancellable.cancel()
-        self.b.basebackup(None, None, {'foo': 'bar'})
+    @patch.object(CancellableSubprocess, 'call')
+    @patch.object(Postgresql, 'data_directory_empty', Mock(return_value=True))
+    def test_basebackup(self, mock_cancellable_subprocess_call):
+        mock_cancellable_subprocess_call.return_value = 0
+        with patch('patroni.postgresql.bootstrap.logger.debug') as mock_debug:
+            self.b.basebackup("", None, {'foo': 'bar'})
+            mock_debug.assert_called_with(
+                'calling: %r',
+                ['pg_basebackup', '--pgdata=data/test0', '-X', 'stream', '--dbname=', '--foo=bar'],
```

Comment:
It fails on windows because path delimiter isn't `'/'`, but `'\'`.
You have to check for `self.p.data_dir` instead of hard-coded string.

Reply:
Fixed